src/shiv/bootstrap/__init__.py

```diff
@@ -140,10 +140,18 @@ def _first_sitedir_index():
             return index
 
 
-def _extend_python_path(environ, additional_paths):
+def _extend_python_path(environ, paths):
+    """Create or extend a PYTHONPATH variable with the frozen environment we are bootstrapping with."""
+
+    # we don't want to clobber any existing PYTHONPATH value, so check for it.
     python_path = environ["PYTHONPATH"].split(os.pathsep) if "PYTHONPATH" in environ else []
-    python_path.extend(additional_paths)
-    environ["PYTHONPATH"] = os.pathsep.join(python_path)
+
+    # extend the supplied paths to include any existing PYTHONPATH
+    paths.extend(python_path)
+
+    # put it back into the environment so that PYTHONPATH contains the shiv-manipulated paths
+    # and any existing PYTHONPATH values.
+    environ["PYTHONPATH"] = os.pathsep.join(set(paths))
```

Comment:
Using `set` is a problem because the order isn't consistent (which is a little surprising, but seems to be the case on 3.6 at least).

Reply:
And having duplicate paths doesn't seem like an issue?